pyam/core.py

```diff
@@ -744,7 +753,9 @@ def rename(self, mapping=None, inplace=False, append=False,
         idx = ret.meta.index.isin(_make_index(ret.data[rows]))
 
         # if `check_duplicates`, do the rename on a copy until after the check
-        _data = ret.data.copy() if check_duplicates else ret.data
+        # _data = ret.data.copy() if check_duplicates else ret.data
+        # TODO reactivate this avoidance of creating a copy
```

Comment:
```suggestion
```

Is this solved?

Reply:
This is not solved, and not all functions are (intended to be) refactored in this PR - in particular this one and the aggregation functions still use the dataframe implementation (always creating a copy of the pandas.Series rather than the more efficient "only create a copy if necessary").

There was a trade-off on how many functions to refactor in one PR - the plan is that @macflo8 and I continue to work on this throughout September and then have a new release.